sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/aggregator/DoubleExponentialHistogramAggregatorTest.java

```diff
@@ -122,6 +135,45 @@ void testInvalidRecording() {
     assertThat(acc.getZeroCount()).isEqualTo(0);
   }
 
+  @ParameterizedTest
+  @MethodSource("provideAggregator")
+  void testRecordingsAtLimits(DoubleExponentialHistogramAggregator aggregator) {
+    AggregatorHandle<ExponentialHistogramAccumulation> aggregatorHandle = aggregator.createHandle();
+
+    aggregatorHandle.recordDouble(Double.MIN_VALUE);
+    aggregatorHandle.recordDouble(Double.MAX_VALUE);
+
+    ExponentialHistogramAccumulation acc = aggregatorHandle.accumulateThenReset(Attributes.empty());
+    List<Long> bucketCounts = Objects.requireNonNull(acc).getPositiveBuckets().getBucketCounts();
+
+    // assert buckets == [1 0 0 0 ... 1]
+    assertThat(bucketCounts.get(0)).isEqualTo(1);
+    assertThat(bucketCounts.get(bucketCounts.size() - 1)).isEqualTo(1);
+    assertThat(bucketCounts.stream().filter(i -> i == 0).count())
+        .isEqualTo(bucketCounts.size() - 2);
+
+    // With 320 buckets, minimum scale is -3
+    assertThat(acc.getScale()).isEqualTo(-3);
+
+    // if scale is -3, base is 256.
+    int base = 256;
+
+    // Verify the rule holds:
+    // base ^ (offset+i) <= (values recorded to bucket i) < base ^ (offset+i+1)
+
+    // lowest bucket:
+    assertThat(Math.pow(base, acc.getPositiveBuckets().getOffset()))
+        .isLessThanOrEqualTo(Double.MIN_VALUE);
+    assertThat(Math.pow(base, acc.getPositiveBuckets().getOffset() + 1))
+        .isGreaterThan(Double.MIN_VALUE);
+
+    // highest bucket
+    assertThat(Math.pow(base, acc.getPositiveBuckets().getOffset() + bucketCounts.size() - 1))
+        .isLessThanOrEqualTo(Double.MAX_VALUE);
+    assertThat(Math.pow(base, acc.getPositiveBuckets().getOffset() + bucketCounts.size()))
```

Comment:
The result of `pow` is `double` so I don't see how it is possible for it to be `greaterThan(Double.MAX_VALUE)`. This assertion seems to be passing though maybe due to some non-obvious overflow happening, but there is probably a better assertion than this non-intuitive one.

Reply:
This test case relies on the result of `Math.pow()` here returning `Infinity`. It's not entirely accurate, but it works and indicates that the upper bound of the bucket is placed correctly above the recorded value. I suppose I could use `BigDecimal` here instead? 

See above comment for problem with `BigDecimal`.